rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -107,9 +109,16 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None, raw=F
         self.topic = topic
         self.msg_class = msg_class
         self.node_handle = node_handle
-        # TODO(@jubeira): add support for other QoS.
+
+        # Some publishers (like /tf_static) publish messages with qos profiles that are incompatible
+        # with the default qos profile. This qos profile indicates we are ok receiving any message types.
+        qos = QoSProfile(
+            depth=10,
+            durability=QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL,
+        )
```

Comment:
The main thing missing is switching between reliable and best-effort based on what is being published. Attempting to open a reliable subscription to a best-effort publisher will fail while opening a best-effort subscription on a reliable publisher could unnecessarily drop messages.

Reply:
Should we use https://github.com/RobotWebTools/rosbridge_suite/pull/654 instead which attempts to mirror the qos profile of the publisher?